sdcm/utils/thread.py

```diff
@@ -0,0 +1,26 @@
+import sys
+from functools import wraps
+import os
+import traceback
+
+
+def raise_event_on_failure(func):
+    """
+    Decorate a function that is running inside a thread,
+    when exception is raised in this function,
+    will raise an Error severity event
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        result = None
+        _test_pid = os.getpid()
+        from sdcm.cluster import Setup
+        from sdcm.sct_events import ThreadFailedEvent
+        try:
+            result = func(*args, **kwargs)
+        except Exception as ex:  # pylint: disable=broad-except
+            Setup.tester_obj().result.addFailure(Setup.tester_obj(), sys.exc_info())
```

Comment:
Why do you add a failure to the tester object?

Reply:
In the previous implementation that was using signal to stop the test immediately, the exception was add to the test as failure or error. 

this mean that when the test fail, you would see this exception as part of the reason the test failed.

I think it's adding a bit more clarity when reading the logs, but also having only the event might be good enough.

Reply:
let's remove this and use Event only (if we want to fail the test we can raise the severity to critical, but let's keep it as error)

Reply:
I think both Error and Critical fail the test.

But I'll remove it anyhow